test/integration/image-component/basic/pages/missing-observer.js

```diff
@@ -1,12 +1,14 @@
-import React, { useLayoutEffect } from 'react'
 import Image from 'next/image'
+import Script from 'next/script'
 
 const Lazy = () => {
-  useLayoutEffect(() => {
-    IntersectionObserver = null //eslint-disable-line
-  })
```

Comment:
Without setting it to null/undefined, how will this test correctly detect the missing IntersectionObserver?

Reply:
> Without setting it to null/undefined, how will this test correctly detect the missing IntersectionObserver?

I replace the `useLayoutEffect` with a `next/script` component.

Reply:
But this test wasn't testing `next/script` before. Its purpose was to test a missing `IntersectionObserver` and lazy loading an image. See https://github.com/vercel/next.js/pull/18345

Reply:
@styfle I am wondering how to test against it then.

https://github.com/vercel/next.js/blob/6f2eb24109592c50074bab18bb67b0f0196a5bae/packages/next/client/use-intersection.tsx#L26

As I mentioned, `hasIntersectionObserver` is calculated before the `useLayoutEffect` is committed. So we just can't dynamically mock it.

Reply:
Hmm, line 26 (`const hasIntersectionObserver`) was always calculated at the top level, so are you suggesting the test was never actually working as intended?

Reply:
> Hmm, line 26 (`const hasIntersectionObserver`) was always calculated at the top level, so are you suggesting the test was never actually working as intended?

During the test locally, I notice that my test failed with an error `IntersectionObserver is not a constructor`.

First I thought maybe caused by `const hasIntersectionObserver = typeof IntersectionObserver !== 'undefined'`, and `typeof null` is `object`. So I change the line to `const hasIntersectionObserver = typeof IntersectionObserver === 'function'`.
But it doesn't work either, then I realize that `useLayoutEffect` would be too late to mock the global.